### DISCUSSION TITLE
Intellisense not working when editing subfiles

### DISCUSSION THREAD

**Discussion by Nayuta7355608** [James-Yu/LaTeX-Workshop#4435] [Q&A]
I'm working on a latex project with a multi-files structure using the `subfiles` package. I have some `.tex` files, which contains some new commands and new environments defined for the project, input in the `main.tex` by the marco `\input`. However, I find that intellisense offers auto-completion for these newly defined commands and citations only in the main file `main.tex`. It offers nothing about these new commands  and citations when I'm editing subfiles, which are included in the `main.tex` using `\subfile`. 
For the reference `\ref` and `\eqref` the intellisense works in the subfiles. The projects structure looks like this:
`main.tex`:
```
\documentclass[A4,12pt,twoside]{book}

\input{mystyle.tex}
\input{theorems.tex} 
\input{mathcommand.tex}
\begin{document}
% %--------------------------------------------------------------------------
% %         Core of the document 
% %--------------------------------------------------------------------------

\part{part1}
\subfile{files/part1/chapter1}
\subfile{files/part1/chapter2}

\part{part2}
\subfile{files/part2/chapter1}
\subfile{files/part2/chapter2}

\nocite{*} % to cite evey things, else cite each on using : \cite{ifrs17}. 
\printbibliography %to print bibliographie

\end{document}
```

subfiles like `chapter1.tex`:
```
% !TEX root = ../main.tex
\documentclass[../main.tex]{subfiles}
\begin{document}
contents
\end{document}
```
I tried adding the announcement `% !TEX root = ../main.tex` in the first line in subfiles, and tried the configuration variables `latex-workshop.latex.texDirs` and `latex-workshop.latex.bibDirs`. None of them makes intellisense work in subfiles

**Comment by James-Yu** [comment]
<img width="637" alt="image" src="https://github.com/user-attachments/assets/750eb5b1-38db-4c50-94e2-b2b5a227e53c">

**Reply by Nayuta7355608** [reply]
Thank you for your reply! I found that I misused `.../main.tex`, for which I should use `../../main.tex`. It should be the reason why intellisense didn't find the root file. It's also interesting that the project could be compiled successfully without correctly setting the path for `main.tex` in subfiles.
